packages/distributor-oracle/src/server.ts

```diff
@@ -444,8 +424,44 @@ export class OracleServer {
       requireAllSignatures: false,
       verifySignatures: false,
     });
+    return {success: true, transaction: serialized};
+  }
+
+  private async signBulkTransactionsHandler(
+    req: FastifyRequest<{ Body: {transactions: number[][] } }>,
+    res: FastifyReply
+  ) {
+    if (!req.body.transactions) {
+      res.status(400).send({ error: "No transactions field" });
+      return;
+    }
+
+    let _this = this;
+    const serializedTxs = await Promise.all(req.body.transactions.map(async (txData) => {
+      const result = await _this.signTransaction(txData);
+      return result.success ? result.transaction : result.message;
```

Comment:
I don't love this mechanism of handling failure. How can I tell on the client side what did succeed vs what didnt? Do I just Transaction.from and catch when it explodes?

Reply:
Yes that was the original idea, so successful transactions in the batch can be executed. I've changed it to return an error from the endpoint now if any of the transactions failed to be signed.